ToolChain/Compiler/Compiler.py

```diff
@@ -0,0 +1,1877 @@
+#!/usr/bin/env python
+#  -*- coding: <utf-8> -*-
+
+"""
+This file is part of Spartacus project
+Copyright (C) 2018  CSE
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+
+from ToolChain.Compiler.Constants import ACCEPTED_TYPES, \
+                                         IGNORE_CHARS, \
+                                         BOOLEAN_OPERATORS, \
+                                         ALLOWED_CHARS, \
+                                         ARRAY_PATTERN, \
+                                         SINGLE_QUOTE, \
+                                         DOUBLE_QUOTE, \
+                                         BINARY_OPERATORS
+
+from ToolChain.Compiler.MathParser import tokenize, \
+                                          infixToPostfix, \
+                                          evaluatePostfix
+
+import re
+
+__author__ = "CSE"
+__copyright__ = "Copyright 2018, CSE"
+__credits__ = ["CSE"]
+__license__ = "GPL"
+__version__ = "3.0"
+__maintainer__ = "CSE"
+__status__ = "Dev"
+
+
+class Compiler:
+    """
+    This is a compiler that converts C code to Capua ASM. It currently supports a small subset of the C programming
+    language, and functionality will progressively be added. It makes use of a finite state machine type model, using
+    states to determine the next expected input, and how to handle the information. Consult the documentation for
+    details on supported features and limitations.
+    """
+
+    state = 0                    # "States" are used to determine our next path for processing the C file
+    currentVar = ""              # Name of variable being evaluated
+    currentType = ""             # Current data type being read, before method/variable declaration
+    currentMethod = ""           # String containing the current method being evaluated
+    expectFlag = 0               # Used to control what input we expect next
+    mathFormula = ""             # Will contain our fully assembled math expressions for variable assignments
+    memoryLocation = 0x40000000  # Memory location for local variables.
+    varList = []                 # Contains a list of variable names
+    varLocation = {}             # Contains the memory location for all variables
```

Comment:
Although I am not done looking/testing this code, I do feel like a relatively large number of these variables don't have to be global to a given object.